### ISSUE TITLE
Feature request: Restart broken Download s

### ISSUE THREAD

**Issue by maxaustria** [nextcloud/android#209]
Right now I am in a rought network environment and maybe 30% of all downloads don't complete successfull. I would love to have an interface simmilar to upload for downloads or integrated both in the same with filters to restart broken downloads. Maybe it would be also helpful to restart all broken uploads or downloads at once.

**Comment by tobiasKaminsky** [comment]
This is the same as #191.
I already have a rough idea how to do it.
Let's track it there.

Restarting all uploads is already possible: swipe left to the upload pane.

**Comment by qdii** [comment]
I might be mistaken, but I think this is different from #191 because:
- restarting broken downloads may be implemented by starting the download over from byte 0.
- chunked downloads doesn't imply auto-restarting.

For context: I’m in vacation in Indonesia, where internet is extremely broken. In addition to very low bandwidth, routers will restart every half an hour here. That means I cannot use Nextcloud to download a movie on my Android tablet. Restarting broken downloads would not help unless it is done by chunks.

**Comment by tobiasKaminsky** [comment]
Auto restarting a download is not that easy as it might get into a loop that consumes much battery.
Or do you have a good idea when to auto restart one (or even many) downloads? 
So I suggest to restart the download manually, but as with chunked download the file will not be downloaded all over again.

**Comment by AndyScherzinger** [comment]
The simplest solution I can think of that would not solve this matter but still help would be to have certain events to trigger a check if up/downloads could/should be restarted:
* (re-)connect to (non-metered) wifi
* app (re-)opened
* account switched

**Comment by tobiasKaminsky** [comment]
Ok, then let us do this, but only after resuming downloads is possible as otherwise we would waste a lot of bandwith.

**Comment by AndyScherzinger** [comment]
Perfectly fine with me!

**Comment by maxaustria** [comment]
I am very happy if I have a list of all broken downloads and I can restart them per item or all at once. So basically same functionality that is there for uploads also for downloads.

Talking about "auto" restart, I don't really like it. Especially when you switch between wifi and mobile data environments this could be more annoying then helpful.

Handling of big files I gave up so far, I was in India, China, Indonesia, a file upload or download currentlygreater 50mb is not working in this environments. It would be really nice to see chunked solutions here in place but that was not my biggest priority when I opened the issue.

**Comment by AndyScherzinger** [comment]
@maxaustria chunked uploads is already implemented and will be shipped with the next release :racehorse:

**Comment by maxaustria** [comment]
A "scenario"... Usually there is nothing like a "good" network connection here. So for example, I am sitting in a restaurant, desperately try to book a train ticket. The restaurant has free WiFi at 2 to 10kb/s. I use firewall profiles right now to avoid any other interference with the network other than my booking app. So if I switch on WiFi and my next cloud client starts making traffic in this scenario in don't like it. Maybe later I come home at my guesthouse, its late in the night so I know the network is better, I connect wifi and then restart all files that failed to download. Until the next morning the download is finished, all is good.

**Comment by AndyScherzinger** [comment]
@maxaustria understood, thanks :) Still then you could run into the same problem, e.g. you are in your guesthouse, restart the downloads @10pm and for some reason the wifi breaks down at 11pm for just a minute. You wake up the next morning, look on your phone and basically nothing has been downloaded. :(

**Comment by maxaustria** [comment]
Usually it's not wifi which is down. It's the DSL line. But both can happen. Or here in China they just cut SSL connection for fun I have the feeling sometimes. I was thinking what would help me the most on "auto restart" and came up with the following idea...

There is a setting "auto restart transfers", so in case you don't want this kind of things you can suppress it. Maybe also if you are on wifi or mobile data might also be important but I saw an other issue about this.

The triggering of the restart should be on various events, beside the obvious (app start, wifi on, ..) I think you need also something like wait 5min, try again for 5 times total. Then go to failed that needs manual handling in order to catch all the DSL line down, ssl connection reset and other scenarios.

But like I said before, if I just get a list of all broken ones, and can restart them all at once I would very happy already :) the auto stuff is nice but maybe not first priority when I look at all the other open topics.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @jancborchardt what do you think? After the implementation of the chunked downloads mechanism, having a downloads view (like the uploads view) would help plus having certain (smart) auto-restart triggers in a later iteration would solve this matter or at least remove most of the pain points.

**Comment by jancborchardt** [comment]
The »Uploads« view should evolve to become an »Activity« view. Uploads, downloads, and in the future even shares and comments. Just like the web interface.

**Comment by xXSTrikeXx** [comment]
#1328 #285

**Comment by AndyScherzinger** [comment]
nope, this is related to downloads thus not related to auto upload

**Comment by matzl** [comment]
I just wasted 2/3 of my mobile data trying to download large video files (200-300 MB) in the Nextcloud App sitting in a train using mobile data. It would be very, very beneficial to improve file synchronization with slow and interrupting internet connection.
In my use case, I sometimes had 4G/4G+, then pauses of 1-10 minutes without internet or only EDGE and then 4G again. This is very common when sitting in a train or a car. The download stopped multiple tines at 50 or even 96%, which is extremely depressing.
